### PR TITLE
trace agent v0.5 integration tests

### PR DESCRIPTION
**Pull request by richardstartin**
will convert to reviewable once the trace agent is released

### CODE REVIEW

**Comment by richardstartin** [comment]
Current failures caused by the agent producing an empty response instead of service rates was acknowledged as a bug in 7.22.0-rc.1 and will need to update to 7.22.0-rc.2 when it's ready.

**Comment by richardstartin** [comment]
The trace agent release candidate now passes our integration tests both with v0.5 enabled and disabled, so I would like a review for the changes here.

File: dd-trace-core/src/main/java/datadog/trace/common/writer/ddagent/DDAgentApi.java [hunk 1]
```diff
@@ -186,28 +199,26 @@ private void countAndLogFailedSend(
       final IOException outer) {
     // count the failed traces
     this.failedTraces += traceCount;
-
     // these are used to catch and log if there is a failure in debug logging the response body
-    IOException exception = outer;
     boolean hasLogged = false;
-
+    String agentError = getResponseBody(response);
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/common/writer/ddagent/DDAgentApi.java:204]: Nice cleanup

**Comment by richardstartin** [comment]
> The timeout fix feels a bit weird. Where is that timeout configured?

I don't know. In fact this whole integration testing module is a bit weird and probably needs cleaning up.

* there are no debug logs for when things fail
* there are comments elsewhere about not testing certain requests because of timeouts
* On other branches, I've changed tests to push failing conditions back a few lines, to find the old assertion is failing on the same line (caching)?
* the module has nonstandard directory structure
* runs testconatainers locally, but depends on a provisioned image in CI
* the testcontainers part wasn't even compiling (see the `Config` -> `ConfigDefaults` change), but the tests were running in CI

I suggest we probably need to start again with integration testing, but this PR just uses what we already have with a new image.

**Comment by bantonsson** [comment]
> I suggest we probably need to start again with integration testing, but this PR just uses what we already have with a new image.

💯

**Comment by tylerbenson** [comment]
Going to merge this so I can do the release.

**Pull request merged**